lib/rom/sql/extensions/postgres/types/range.rb

```diff
@@ -6,7 +6,19 @@ module ROM
   module SQL
     module Postgres
       module Values
-        Range = ::Struct.new(:lower, :upper, :lower_bound, :upper_bound)
+        Range = ::Struct.new(:lower, :upper, :bounds) do
+          def initialize(lower, upper, bounds = :'[)')
+            super
+          end
+
+          def exclude_begin?
+            bounds[0] == '('
+          end
+
+          def exclude_end?
+            bounds[1] == ')'
+          end
```

Comment:
It would be nice to avoid allocation of these strings and use constants instead, like `PAREN_LEFT = '('.freeze` and `PARENT_RIGHT = ')'.freeze`.